crypto/http/http_client.c

```diff
@@ -1106,17 +1119,24 @@ int OSSL_HTTP_proxy_connect(BIO *bio, const char *server, const char *port,
     char *mbuf = OPENSSL_malloc(BUF_SIZE);
     char *mbufp;
     int read_len = 0;
-    int rv;
     int ret = 0;
     BIO *fbio = BIO_new(BIO_f_buffer());
+#ifndef OPENSSL_NO_SOCK
+    int rv;
     time_t max_time = timeout > 0 ? time(NULL) + timeout : 0;
+    int issocket;
+#endif
 
     if (bio == NULL || server == NULL || port == NULL
             || (bio_err != NULL && prog == NULL)) {
         HTTPerr(0, ERR_R_PASSED_NULL_PARAMETER);
         goto end;
     }
 
+#ifndef OPENSSL_NO_SOCK
+    issocket = BIO_get_fd(bio, NULL) > 0;
```

Comment:
@mattcaswell, were you sure to derive `issocket` from `bio`?
I think it should be from `fbio` instead.

Reply:
It shouldn't really make a difference. fbio is just a filter bio so should pass the BIO_C_GET_FD ctrl onto the underlying bio. Either should work.

Reply:
Of course if you  do it against fbio you have to do it *after* the `BIO_push(fbio, bio)` call below.

Reply:
Thanks for your quick response.
Meanwhile I think I found an even better solution that saves callers of `BIO_wait()` from worrying whether the BIO is a socket or not and whether sockets are supported.